libs/components/src/alert/vl-alert.component.cy.ts

```diff
@@ -0,0 +1,139 @@
+    });
+
+    it('should be removed after clicking the close button and send a VlAlertClosedEvent', () => {
+        cy.createStubForEvent('vl-alert', VlAlertClosedEvent.eventType);
+        cy.get('vl-alert').shadow().find('#close').click();
+        cy.get('vl-alert').should('not.exist');
+        cy.get('@' + VlAlertClosedEvent.eventType).should('have.been.calledOnce');
+    });
+});
+
+describe('component vl-alert naked', () => {
+    beforeEach(() => {
+        cy.mount(html`
+            <vl-alert
+                data-cy="alert"
+                data-vl-icon="warning"
+                data-vl-title="Lorem ipsum"
+                data-vl-size=""
+                data-vl-type="error"
+                data-vl-closable=""
+                data-vl-naked="true"
+                data-vl-message="Phasellus congue ipsum ut felis auctor, eget maximus justo dapibus. Nam sit amet pulvinar odio. Maecenas rhoncus quam eget neque porttitor, et faucibus nisl elementum."
+            >
+            </vl-alert>
+        `);
+    });
+    it('should mount', () => {
+        cy.get('vl-alert');
+    });
+
+    it('should be accessible', () => {
+        cy.injectAxe();
+
+        cy.checkA11y('vl-alert');
+    });
+
+    it('should contain a marked title', () => {
+        cy.get('vl-alert')
+            .shadow()
+            .find('#alert')
+            .should('have.class', 'vl-alert')
+            .find('slot')
+            .should('have.class', 'vl-u-mark--error')
+            .contains('Lorem ipsum');
+    });
+
+    it('should contain marked text', () => {
+        cy.get('vl-alert')
+            .shadow()
+            .find('#alert #message > p')
+            .should('have.class', 'vl-u-mark--error')
+            .contains(
+                'Phasellus congue ipsum ut felis auctor, eget maximus justo dapibus. Nam sit amet pulvinar odio. Maecenas rhoncus quam eget neque porttitor, et faucibus nisl elementum.'
+            );
+    });
+
+    it('should change the title marking', () => {
+        cy.get('vl-alert').invoke('attr', 'data-vl-type', 'success');
+        cy.get('vl-alert')
+            .shadow()
+            .find('#alert slot')
+            .should('have.class', 'vl-u-mark--success')
+            .contains('Lorem ipsum');
+    });
+
+    it('should change the text marking', () => {
+        cy.get('vl-alert').invoke('attr', 'data-vl-type', 'success');
+        cy.get('vl-alert')
+            .shadow()
+            .find('#alert #message > p')
+            .should('have.class', 'vl-u-mark--success')
+            .contains(
+                'Phasellus congue ipsum ut felis auctor, eget maximus justo dapibus. Nam sit amet pulvinar odio. Maecenas rhoncus quam eget neque porttitor, et faucibus nisl elementum.'
+            );
+    });
+
+    it('should contain an icon', () => {
+        cy.get('vl-alert').shadow().find('.vl-alert__icon > span').should('have.attr', 'data-vl-icon', 'warning');
+    });
+});
```

Comment:
eventueel nog tests toevoegen voor het controleren dat de slots werken?

als je dit niet nuttig vind, kan je dit laten - op zich gebeuren er geen gekke dingen met de slots in dit component

Reply:
Tests toegevoegd